test/blackbox/api/dds-pim/PubSubParticipant.hpp

```diff
@@ -228,8 +228,13 @@ class PubSubParticipant
     {
         if (enable_datasharing)
         {
+#ifdef ANDROID
+            datareader_qos_.data_sharing().automatic("/data/local/tmp/");
+            datawriter_qos_.data_sharing().automatic("/data/local/tmp/");
```

Comment:
This is hardcoded. Should we provide a mechanism to change it?

Applies to all the test files that have been modified

Reply:
Calls to `.automatic()` can be left without an argument. They will automagically use `BOOST_INTERPROCESS_SHARED_DIR_PATH`, which we have defined to `/data/local/tmp` on #2977